Libs/MRML/Core/Testing/vtkMRMLDoubleArrayNodeTest1.cxx

```diff
@@ -10,18 +10,413 @@
 
 =========================================================================auto=*/
 
+// MRML includes
+#include "vtkMRMLCoreTestingMacros.h"
 #include "vtkMRMLDoubleArrayNode.h"
+#include "vtkMRMLDoubleArrayStorageNode.h"
 
+// VTK includes
+#include <vtkNew.h>
 
-#include "vtkMRMLCoreTestingMacros.h"
+// STD includes
+#include <vector>
+#include <sstream>
+
+//---------------------------------------------------------------------------
+bool TestSetValues();
+bool TestSetValue();
+bool TestSetXYValue();
+bool TestSetXYerrValue();
+bool TestAddValues();
+bool TestAddValue();
+bool TestAddXYValue();
+bool TestAddXYerrValue();
+bool TestReadFileWithLabels(std::string filepath);
+bool TestReadFileWithoutLabels(std::string filepath);
+bool TestReadOldFile(std::string filepath);
+
+//---------------------------------------------------------------------------
+int vtkMRMLDoubleArrayNodeTest1(int argc, char * argv[])
+{
+  vtkNew<vtkMRMLDoubleArrayNode> node1;
+
+  EXERCISE_BASIC_OBJECT_METHODS(node1.GetPointer());
+
+  EXERCISE_BASIC_MRML_METHODS(vtkMRMLDoubleArrayNode, node1.GetPointer());
+
+  if (argc != 2)
+    {
+    std::cerr << "Line " << __LINE__
+	      << " - Missing parameters !\n"
+	      << "Usage: " << argv[0] << " /path/to/temp"
+	      << std::endl;
+    return EXIT_FAILURE;
+    }
+
+  const char* tempDir = argv[1];
+
+  std::string fileNameOldFile = std::string(tempDir) + "/vtkMRMLDoubleArrayNodeTest1_OldFile.mcsv";
+  std::string fileNameWithLabels = std::string(tempDir) + "/vtkMRMLDoubleArrayNodeTest1_WithLabels.mcsv";
+  std::string fileNameWithoutLabels = std::string(tempDir) + "/vtkMRMLDoubleArrayNodeTest1_WithoutLabels.mcsv";
+  
+  bool res = true;
+  res = res && TestSetValues();
+  res = res && TestSetValue();
+  res = res && TestSetXYValue();
+  res = res && TestSetXYerrValue();
+  res = res && TestAddValues();
+  res = res && TestAddValue();
+  res = res && TestAddXYValue();
+  res = res && TestAddXYerrValue();
+  res = res && TestReadFileWithLabels(fileNameWithLabels);
+  res = res && TestReadFileWithoutLabels(fileNameWithoutLabels);
+  res = res && TestReadOldFile(fileNameOldFile);
+  
+  return res ? EXIT_SUCCESS : EXIT_FAILURE;
+}
+
+//---------------------------------------------------------------------------
+bool doubleArrayMatch(double array1[], double array2[], unsigned int sizeOfArray)
+{
+  for(unsigned int i = 0; i < sizeOfArray; i++)
+    {
+    std::ostringstream s1;
+    std::ostringstream s2;
+    s1 << array1[i];
+    s2 << array2[i];
+    if( s1.str() != s2.str() )
+      {
+      std::cout << "ERROR : " << array1[i] << " not equal to " << array2[i] << std::endl;
```

Comment:
that you would use here to do something like: 

```
std::cerr << "Line " << line << " - Array do not match. " << array1[i] << " not equal to " << array2[i] << std::endl
```

Consider using `std::cerr` instead of `std::cout`